### PR TITLE
update BYOC documentation with AWS details

### PR DESCRIPTION
**Pull request by nicolai86**
Update BYOC documentation

## What does this PR do?

This PR adds details that _must_ be public for customers to be able to create their own BYOC clusters. 
They are technical in nature but since they are prerequisites for BYOC they must be documented.

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: docs/source-app/workflows/byoc/create_cluster.rst [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+
+To start a cluster, Lightning will need the external-id you set in step 1, and the ARN role. Click on your new role to find them (note: you need the ARN listed at the top of the page, not the one in “trusted entitles”).
+
+
+Record the ARN and the external ID. You’ll need them for your next step.
+
+-----
+
+*******************************
+Create a Lightning BYOC cluster
```
Comment (awaelchli) [inline docs/source-app/workflows/byoc/create_cluster.rst:72]: Is this step 3? it is the same header level as step 2

File: docs/source-app/workflows/byoc/run_on_cluster.rst [hunk 2]
```diff
@@ -0,0 +1,27 @@
+:orphan:
+
+.. _run_on_cluster:
+
+
+*********************************
+Run app on your Lightning Cluster
+*********************************
+
+Once your cluster is running, you can run any Lightning App on your cluster. To run an App on the Lightning BYOC cluster, use --cloud –cluster-id <cluste-id> in the command:
```
Comment (awaelchli) [inline docs/source-app/workflows/byoc/run_on_cluster.rst:10]:
```suggestion
Once your cluster is running, you can run any Lightning App on your cluster. To run an App on the Lightning BYOC cluster, use ``--cloud –cluster-id <cluste-id>`` in the command:
```

File: docs/source-app/workflows/byoc/aws_console.rst [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+###########################################
+Create AWS role with AWS management console
+###########################################
+
+See the following video for instructions:
+
+.. raw:: html
+
+    <video width="100%" max-width="800px" controls autoplay muted playsinline
+    src=" https://lightningaidev.wpengine.com/wp-content/uploads/2022/12/byoc.mp4"></video>
```
Comment (awaelchli) [inline docs/source-app/workflows/byoc/aws_console.rst:15]:
```suggestion
    src="https://lightningaidev.wpengine.com/wp-content/uploads/2022/12/byoc.mp4"></video>
```

File: docs/source-app/workflows/byoc/run_on_cluster.rst [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+.. code:: bash
+
+   lightning run app app.py  —-cloud —-cluster-id <cluster-id>
```
Comment (awaelchli) [inline docs/source-app/workflows/byoc/run_on_cluster.rst:14]:
```suggestion
   lightning run app app.py --cloud --cluster-id <cluster-id>
```

File: docs/source-app/workflows/byoc/run_on_cluster.rst [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+Here’s an example:
+
+.. code:: bash
+
+   lightning run app app.py –cloud —-cluster-id my-byoc-cluster
```
Comment (awaelchli) [inline docs/source-app/workflows/byoc/run_on_cluster.rst:20]:
```suggestion
   lightning run app app.py --cloud --cluster-id my-byoc-cluster
```

File: docs/source-app/workflows/byoc/create_cluster.rst [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+*******************************
+
+Now that you have created a role and policy on AWS, you can start creating Lightning clusters.
+
+Create a Lightning BYOC cluster using the following command:
+
+.. code:: bash
+
+   lightning create cluster <cluster-name>  --role-arn <ARN> --external-id <EXTERNAL-ID>
```
Comment (awaelchli) [inline docs/source-app/workflows/byoc/create_cluster.rst:81]:
```suggestion
   lightning create cluster <cluster-name> --role-arn <ARN> --external-id <EXTERNAL-ID>
```

**Review by awaelchli — APPROVED** [review]

File: docs/source-app/workflows/byoc/create_cluster.rst (hunk 1 above)
Comment (nicolai86) [reply docs/source-app/workflows/byoc/create_cluster.rst:72]: Yes, this is Step 3. let me change that.

**Review by Borda — APPROVED** [review]

**Pull request merged**